lib/smart_answer/helpers/format_capture_helper.rb

```diff
@@ -0,0 +1,64 @@
+module SmartAnswer
+  module Helpers::FormatCaptureHelper
+    DEFAULT_FORMATS = {
+      govspeak: [/^body$/, /^post_body$/, /^next_steps$/],
+      text: [/^title$/, /^meta_description$/, /^hint$/, /^label$/, /^suffix_label$/, /^error_*./],
+    }.freeze
+
+    def content_for(name, content = nil, options = {}, &block)
+      if block_given?
```

Comment:
This confused me until I found [the interface we're trying to conform to](https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/actionview/lib/action_view/helpers/capture_helper.rb#L155). We should comment about that.

I think we should also match the normal behaviour of the interface, so this works with passed content; although it's not required, it confused me when trying to understand how this method is used.

Reply:
I'm going to try and rename this, rather than override `content_for` - should also help simplify the code as we don't have worry about the different ways `content_for` gets used.

Reply:
So I've refactored it into a new method, not overriding content_for so hopefully that clarifies things a bit and removes the dependency to match the behaviour.

Reply:
Yes, that's much easier to understand. Do we still need the `if` check? I think we still have a semantic problem that this is only one half of the normal `content_for` method, which stores and retrieves content to render.

The new name is (therefore) confusing for me, since we're not actually `render`ing anything where the method is called; instead, the block is captured and a separate call must be made later to actually do the rendering onto the page.

The only name I could think of that was less ambiguous was `format_content_for`, since it can only apply to content going "in", and doesn't make sense for rendering "out" e.g. `<%= format_content_for :body %>`.

I know it sounds like a minor issue. Since this method is used all over the app, I think it makes sense to have a little more discussion on what it's called, to see if we can avoid confusing people in the future, who won't know about the helper initially.

Reply:
Regarding [your main comment](https://github.com/alphagov/smart-answers/pull/4502#issuecomment-624840109), I'd be content for this to go into another PR.

Reply:
Yup - removed the `if block_given?` line.

So there might be a mis-understand, the `render_content_for` method does "render" (convert govspeak into HTML and wrap in the tags, and call any other erb render functions etc...) the content before storing to be accessed later by `content_for` - thats done on line 12 of the method.

Reply:
> Yup - removed the if block_given? line.

👍 

> So there might be a mis-understand

I think I didn't express myself well. The name of the method implies it returns something, but that's not actually true. It's really a "setter". On balance, the original `content_for` is sufficiently unintuitive that there's not a lot we can do in this space.

Even `format_content_for` is still vague. My concern is that it could be easily confused with the Rails `content_for` method. Something completely different, like `smartanswer_block` would avoid that, since it's clearly specific to the app.

However, none of these suggestions accurately convey what the method does (that's also different to the Rails one), so I'll just resolve this suggestion.